src/main/java/com/softserve/academy/spaced/repetition/DTO/impl/CourseCommentLinkDTO.java

```diff
@@ -0,0 +1,38 @@
+package com.softserve.academy.spaced.repetition.DTO.impl;
+
+import com.softserve.academy.spaced.repetition.DTO.DTO;
+import com.softserve.academy.spaced.repetition.controller.CourseCommentController;
+import com.softserve.academy.spaced.repetition.domain.CourseComment;
+import org.springframework.hateoas.Link;
+
+import java.util.Date;
+
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.methodOn;
+
+public class CourseCommentLinkDTO extends DTO<CourseComment> {
+
+    public CourseCommentLinkDTO(CourseComment courseComment, Link link) {
+        super(courseComment, link);
+        add(linkTo(methodOn(CourseCommentController.class).getAllCommentsByCourse(getEntity().getCourse().getCategory()
+                .getId(), getEntity().getCourse().getId())).withRel("comments"));
```

Comment:
What's the difference between CourseCommentDTO and CourseCommentLinkDTO. Class CourseCommentLinkDTO what is a comment has children comments?

Reply:
CourseCommentLinkDTO has an additional link to the array of all comments for the course

Reply:
Can we extend one from another?

Reply:
I decided to delete CourseCommentLinkDTO, because an additional link to the array of all comments for the course will not be used.